### PR TITLE
add controller,mapper,repository imageuser

### PR DESCRIPTION
**Pull request by Cheephot**

### CODE REVIEW

File: src/ImageService.Business/Commands/ImageUser/GetImageUserCommand.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        public OperationResultResponse<ImageUserResponse> Execute(Guid parentId)
+        {
+            OperationResultResponse<ImageUserResponse> response = new();
+
+            DbImagesUser dbImageUser = _imageUserRepository.Get(parentId);
+
+            if (dbImageUser == null)
+            {
+                response.Body = null;
+                response.Errors.Add("Image was not found.");
```
Comment (ndoriomedov) [inline src/ImageService.Business/Commands/ImageUser/GetImageUserCommand.cs:38]: add response.Status

Comment (Cheephot) [reply src/ImageService.Business/Commands/ImageUser/GetImageUserCommand.cs:39]: done

File: src/ImageService/Startup.cs [hunk 2]
```diff
@@ -77,6 +84,11 @@ public void ConfigureServices(IServiceCollection services)
                 })
                 .AddNewtonsoftJson();
 
+            services.AddTransient<IGetImageUserCommand, GetImageUserCommand>();
```
Comment (BelehovEgor) [inline src/ImageService/Startup.cs:87]: use auto inject

File: src/ImageService.Models.Dto/Responses/ImageDataResponse.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace LT.DigitalOffice.ImageService.Models.Dto.Responses.User
+{
+    public class ImageDataResponse
```
Comment (BelehovEgor) [inline src/ImageService.Models.Dto/Responses/ImageDataResponse.cs:5]: maybe better name is 'ImageResponse' or 'ImageInfo'? we often use word Indo for Dto models

Comment (Cheephot) [reply src/ImageService.Models.Dto/Responses/ImageDataResponse.cs:5]: I think we can make this a separate task for all tables

File: src/ImageService/Startup.cs (hunk 2 above)
Comment (Cheephot) [reply src/ImageService/Startup.cs:87]: I done

File: src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        private readonly IDbImageUserMapper _mapper;
+
+        public CreateImagesUserConsumer(IImageUserRepository repository, IDbImageUserMapper mapper)
+        {
+            _repository = repository;
+            _mapper = mapper;
+        }
+        public async Task Consume(ConsumeContext<ICreateImagesUserRequest> context)
+        {
+            var response = OperationResultWrapper.CreateResponse(CreateImages, context.Message);
```
Comment (MariyaSin) [inline src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs:26]: do not use var

File: src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+            foreach (CreateImageData x in createImage)
+            {
+                images.AddRange(CreateHQ(x));
+            }
+
+            return ICreateImagesResponse.CreateObj(_repository.Create(images));
+        }
+
+        public List<DbImagesUser> CreateHQ(CreateImageData createImageData)
```
Comment (MariyaSin) [inline src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs:44]: HQ- what is it mean?

File: src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        {
+            List<DbImagesUser> db = new();
```
Comment (MariyaSin) [inline src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs:46]: dbImageUsers, not db

File: src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+            var highQ = _mapper.Map(createImageData);
```
Comment (MariyaSin) [inline src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs:47]: do not user var

File: src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+            db.Add(highQ);
+            var createImageData1 = new CreateImageData(createImageData.Name, "преобразованная картинка", createImageData.Extension, createImageData.CreatedBy);
```
Comment (MariyaSin) [inline src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs:49]: do not use var, do not use Russian text in c# code

File: src/ImageService.Models.Dto/Responses/ImageDataResponse.cs (hunk 3 above)
Comment (MariyaSin) [inline src/ImageService.Models.Dto/Responses/ImageDataResponse.cs:5]: rewrite to record

File: src/ImageService.Data/Interfaces/IImageUserRepository.cs [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+using LT.DigitalOffice.Kernel.Attributes;
+using System;
+using System.Collections.Generic;
+
+namespace LT.DigitalOffice.ImageService.Data.Interfaces
+{
+    [AutoInject]
+    public interface IImageUserRepository
+    {
+        List<Guid> Create(List<DbImagesUser> imagesUsers);
```
Comment (MariyaSin) [inline src/ImageService.Data/Interfaces/IImageUserRepository.cs:11]: add empty line

File: src/ImageService.Data/Interfaces/IImageUserRepository.cs [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+        List<DbImagesUser> Get(List<Guid> imageIds);
```
Comment (MariyaSin) [inline src/ImageService.Data/Interfaces/IImageUserRepository.cs:12]: add empty line

File: src/ImageService.Data/Interfaces/IImageUserRepository.cs [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+        DbImagesUser Get(Guid imageId);
```
Comment (MariyaSin) [inline src/ImageService.Data/Interfaces/IImageUserRepository.cs:13]: add empty line

File: src/ImageService.Data/ImageUserRepository.cs [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+        public List<DbImagesUser> Get(List<Guid> imageIds)
+        {
+            return _provider.ImagesUsers.Where(x => imageIds.Contains(x.Id)).ToList();
+        }
+
+        public DbImagesUser Get(Guid imageId)
+        {
+            return _provider.ImagesUsers.Where(x => x.Id == imageId).FirstOrDefault();
+        }
+
```
Comment (MariyaSin) [inline src/ImageService.Data/ImageUserRepository.cs:54]: remove empty line

File: src/ImageService.Data/ImageUserRepository.cs [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+        public List<DbImagesUser> Get(List<Guid> imageIds)
+        {
+            return _provider.ImagesUsers.Where(x => imageIds.Contains(x.Id)).ToList();
+        }
+
+        public DbImagesUser Get(Guid imageId)
+        {
+            return _provider.ImagesUsers.Where(x => x.Id == imageId).FirstOrDefault();
```
Comment (MariyaSin) [inline src/ImageService.Data/ImageUserRepository.cs:52]: _provider.ImagesUsers.FirstOrDefault(x => x.Id == imageId)

File: src/ImageService.Broker/Consumers/ImageUser/GetImagesUserConsumer.cs [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+        public async Task Consume(ConsumeContext<IGetImagesUserRequest> context)
+        {
+            var response = OperationResultWrapper.CreateResponse(GetImages, context.Message);
+
+            await context.RespondAsync<IOperationResult<IGetImagesResponse>>(response);
+        }
+
+        private object GetImages(IGetImagesUserRequest request)
+        {
+            var dbImages = _repository.Get(request.ImageIds);
```
Comment (MariyaSin) [inline src/ImageService.Broker/Consumers/ImageUser/GetImagesUserConsumer.cs:30]: do not use var

File: src/ImageService.Data/ImageUserRepository.cs [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+                return null;
+            }
+
+            _provider.ImagesUsers.AddRange(imagesUsers);
+            _provider.Save();
+
+            return imagesUsers.Select(x => x.Id).ToList();
+        }
+
+        public bool Delete(List<DbImagesUser> imagesUsers)
```
Comment (MariyaSin) [inline src/ImageService.Data/ImageUserRepository.cs:32]: Why you can not pass to the method list of imageIds?

File: src/ImageService.Broker/Consumers/ImageUser/DeleteImagesUserConsumer.cs [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+        public async Task Consume(ConsumeContext<IDeleteImagesUserRequest> context)
+        {
+            var response = OperationResultWrapper.CreateResponse(DeleteImages, context.Message);
+
+            await context.RespondAsync<IOperationResult<bool>>(response);
+        }
+
+        private object DeleteImages(IDeleteImagesUserRequest request)
+        {
+            List<DbImagesUser> dbImages = _repository.Get(request.ImageIds);
```
Comment (MariyaSin) [inline src/ImageService.Broker/Consumers/ImageUser/DeleteImagesUserConsumer.cs:29]: throw to the delet method ids, not dbImages

File: src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+        public async Task Consume(ConsumeContext<ICreateImagesUserRequest> context)
+        {
+            var response = OperationResultWrapper.CreateResponse(CreateImages, context.Message);
+
+            await context.RespondAsync<IOperationResult<ICreateImagesResponse>>(response);
+        }
+
+        private object CreateImages(ICreateImagesUserRequest request)
+        {
+            List<CreateImageData> createImage = request.CreateImagesData;
```
Comment (MariyaSin) [inline src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs:33]: why are you creating this variable? you can use request.CreateImagesData

File: src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+            List<DbImagesUser> images = new();
+
+            foreach (CreateImageData x in createImage)
+            {
+                images.AddRange(CreateHQ(x));
```
Comment (MariyaSin) [inline src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs:38]: a very strange method. Every iteration, you turn to the method that creates the list and then put this list into the list

File: src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs (hunk 7 above)
Comment (Cheephot) [reply src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs]: done

File: src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs (hunk 8 above)
Comment (Cheephot) [reply src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs]: done

File: src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs (hunk 18 above)
Comment (Cheephot) [reply src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs]: Replaced with the normal method

File: src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs (hunk 6 above)
Comment (Cheephot) [reply src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs]: done

File: src/ImageService.Models.Dto/Responses/ImageDataResponse.cs (hunk 3 above)
Comment (Cheephot) [reply src/ImageService.Models.Dto/Responses/ImageDataResponse.cs:5]: done

File: src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs (hunk 4 above)
Comment (Cheephot) [reply src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs:30]: done

File: src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs (hunk 5 above)
Comment (Cheephot) [reply src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs]: done

File: src/ImageService.Data/Interfaces/IImageUserRepository.cs (hunk 10 above)
Comment (Cheephot) [reply src/ImageService.Data/Interfaces/IImageUserRepository.cs:12]: done

File: src/ImageService.Data/Interfaces/IImageUserRepository.cs (hunk 11 above)
Comment (Cheephot) [reply src/ImageService.Data/Interfaces/IImageUserRepository.cs:13]: done

File: src/ImageService.Data/Interfaces/IImageUserRepository.cs (hunk 9 above)
Comment (Cheephot) [reply src/ImageService.Data/Interfaces/IImageUserRepository.cs:11]: done

File: src/ImageService.Data/Interfaces/IImageUserRepository.cs (hunk 11 above)
Comment (Cheephot) [reply src/ImageService.Data/Interfaces/IImageUserRepository.cs:13]: done

File: src/ImageService.Data/Interfaces/IImageUserRepository.cs (hunk 10 above)
Comment (Cheephot) [reply src/ImageService.Data/Interfaces/IImageUserRepository.cs:12]: done

File: src/ImageService.Data/Interfaces/IImageUserRepository.cs (hunk 9 above)
Comment (Cheephot) [reply src/ImageService.Data/Interfaces/IImageUserRepository.cs:11]: done

File: src/ImageService.Data/ImageUserRepository.cs (hunk 12 above)
Comment (Cheephot) [reply src/ImageService.Data/ImageUserRepository.cs:54]: done

File: src/ImageService.Data/ImageUserRepository.cs (hunk 13 above)
Comment (Cheephot) [reply src/ImageService.Data/ImageUserRepository.cs:52]: done

File: src/ImageService.Broker/Consumers/ImageUser/GetImagesUserConsumer.cs (hunk 14 above)
Comment (Cheephot) [reply src/ImageService.Broker/Consumers/ImageUser/GetImagesUserConsumer.cs:30]: done

File: src/ImageService.Data/ImageUserRepository.cs (hunk 15 above)
Comment (Cheephot) [reply src/ImageService.Data/ImageUserRepository.cs]: done

File: src/ImageService.Broker/Consumers/ImageUser/DeleteImagesUserConsumer.cs (hunk 16 above)
Comment (Cheephot) [reply src/ImageService.Broker/Consumers/ImageUser/DeleteImagesUserConsumer.cs]: done

File: src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs (hunk 17 above)
Comment (Cheephot) [reply src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs]: done

**Review by VLiamin — APPROVED** [review]

**Review by ndoriomedov — APPROVED** [review]

**Review by MariyaSin — APPROVED** [review]

**Review by BelehovEgor — APPROVED** [review]

**Pull request merged**